### DISCUSSION TITLE
Control max depth for sidebar that includes autosummary content

### DISCUSSION THREAD

**Discussion by adamcunnington-mlg** [pradyunsg/furo#289] [Potential Issue] [answered]
Thanks for the great theme!

My sidebar is built like this:
```rst
.. toctree::
    :hidden:

    Home <self>
    API Reference <api>
```

but `<api>` contains:
```rst
.. autosummary::
   :recursive:
   :template: module.rst
   :toctree: generated

   bqtemplatemanager
```

The problem is that the sidebar can then be very deep - and `:maxdepth:` doesn't seem to do anything. The result is ugly stlying:
![image](https://user-images.githubusercontent.com/93528462/139953423-da5d8c71-de21-46cf-8c7a-6c806d4b64e1.png)

**Comment by pradyunsg** [comment]
Got a reproducer or sample website for me to look at?

**Reply by adamcunnington-mlg** [reply]
@pradyunsg sure; https://medialabgroup.gitlab.io/apollo/bq-template-manager/generated/bqtemplatemanager.html

**Reply by pradyunsg** [reply, marked as answer]
Furo intentionally presents the entire site structure provided to it -- to avoid having "hidden" parts to the documentation site.

If you still want to exclude auto-generated content from the toctree, I think the right tool for that is https://github.com/executablebooks/sphinx-remove-toctrees.

**Reply by adamcunnington-mlg** [reply]
@pradyunsg got it - but do you think you could improve the styling? As you saw from the site, long links overlap with the arrow dropdown. When you hover over the dropdown, they are truncated which looks much nicer. Perhaps you can style so this behaviour happens by default, maybe using an ellipsis to indicate extra content is hidden?

I don't want to hide the auto-generated content from the sidebar, I just want it to render nicer.

**Reply by pradyunsg** [reply]
Done in fcd6ff66f45f6fb66d26428fe01e9b4a216ffed2. :)

**Reply by KelSolaar** [reply]
Text ellipsis can be done with a `custom.css` file with the following content:


```css
.sidebar-tree .reference {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    word-break: normal;
}
```

**Reply by pradyunsg** [reply]
Not hiding page titles behind ellipsis is intentional.

**Reply by ailin-nemui** [reply]
for the specific case visible in the screen shot, it could also be worth to check the sphinxext-autox module, it can "cut off" the components that are naturally induced from the parent
